### PR TITLE
[node] Use `NodejsLambda` class

### PR DESCRIPTION
**Pull request by TooTallNate**
Updates `@vercel/node` to utilize the `NodejsLambda` class so that it doesn't need to explicitly bundle in the launcher/bridge files into the build result. This reduces the complexity of the Builder and is a cleaner separation of concerns. The `helpers.test.ts` file has been moved to `@vercel/node-bridge` so it's being removed in this PR as well.

### CODE REVIEW

File: packages/node/test/helpers.test.js [hunk 1]
```diff
@@ -1,842 +0,0 @@
-const fetch = require('node-fetch');
-const listen = require('test-listen');
-const qs = require('querystring');
-
-const { createServerWithHelpers } = require('../dist/helpers');
```
Comment (styfle) [inline packages/node/test/helpers.test.js:5]: Lets make sure these tests are moved to the bridge package so we don't lose them

**Review by styfle — APPROVED** [review]

File: packages/node/src/index.ts [hunk 2]
```diff
@@ -342,18 +333,14 @@ export * from './types';
...
   workPath,
   repoRootPath,
   config = {},
   meta = {},
-}: BuildOptions) {
-  const shouldAddHelpers = !(
-    config.helpers === false || process.env.NODEJS_HELPERS === '0'
-  );
-
+}) => {
```
Comment (EndangeredMassa) [inline packages/node/src/index.ts:343]: Should the argument still be typed `BuildOptions`?

**Review by EndangeredMassa — APPROVED** [review]

File: packages/node/test/helpers.test.js (hunk 1 above)
Comment (TooTallNate) [reply packages/node/test/helpers.test.js:5]: Already happened in #7451.

File: packages/node/src/index.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/node/src/index.ts:343]: It is still typed as such due to the `const build: BuildV3` type.

Comment (EndangeredMassa) [reply packages/node/src/index.ts:343]: Ahh, so it is. I saw that it was `any` locally because I had to `yarn build` to get the updated types across.

**Pull request merged**